backend/db/utils.py

```diff
@@ -702,21 +719,23 @@ def reset_temp_refresh_tables(schema: str = SCHEMA):
     print('Error occurred during table refresh. Resetting tables to pre-refresh state; restoring backups.',
           file=sys.stderr)
     with get_db_connection(schema=schema) as con:
-        tables: List[str] = list_tables(con, schema)
-
-        backed_up_tables = [t for t in tables if t.endswith('_old')]
-        for table in backed_up_tables:
-            table = table.replace('_old', '')
-            run_sql(con, f'DROP TABLE IF EXISTS {schema}.{table};')
-            run_sql(con, f'ALTER TABLE {schema}.{table}_old RENAME TO {table};')
-
-        dangling_new_tables = [t for t in tables if t.endswith('_new')]
-        for table in dangling_new_tables:
-            table = table.replace('_new', '')
-            if table in tables:
-                run_sql(con, f'DROP TABLE {schema}.{table}_new;')
-            else:
-                run_sql(con, f'ALTER TABLE {schema}.{table}_new RENAME TO {table};')
+        for item_type, func in (('VIEW', list_views), ('TABLE', list_tables)):
```

Comment:
## The bugfix
I was originally going to solve this with `CASCADE`, which would've probably been fine. However, I needed to fetch view names in order to rename them `e.g. remove `_old`, and since I had the list of views, I felt it'd be safer to just remove these views first before tables, which obsoleted the need for cascade.

This ended up in a nice refactor as well.